src/partest/scala/tools/partest/nest/Runner.scala

```diff
@@ -552,25 +545,30 @@ class Runner(val testInfo: TestInfo, val suiteRunner: AbstractRunner) {
 
   def runPosTest(): TestState =
     if (checkFile.exists) genFail("unexpected check file for pos test (use -Werror with neg test to verify warnings)")
-    else runTestCommon()
+    else runTestCommon()()
 
-  def runNegTest(): TestState = runInContext {
+  def runNegTest(): TestState = {
     // pass if it checks and didn't crash the compiler
     // or, OK, we'll let you crash the compiler with a FatalError if you supply a check file
-    def checked(r: CompileRound) = r.result match {
+    def checked(r: TestState) = r match {
       case s: Skip => s
       case crash @ Crash(_, t, _) if !checkFile.canRead || !t.isInstanceOf[FatalError] => crash
       case _ => diffIsOk
     }
 
-    compilationRounds(testFile).find(r => !r.result.isOk || r.result.isSkipped).map(checked).getOrElse(genFail("expected compilation failure"))
+    runTestCommon(checked, expectCompile = false)()
   }
 
   // run compilation until failure, evaluate `andAlso` on success
-  def runTestCommon(andAlso: => TestState = genPass()): TestState = runInContext {
-    // DirectCompiler already says compilation failed
-    val res = compilationRounds(testFile).find(r => !r.result.isOk || r.result.isSkipped).map(_.result).getOrElse(genPass())
-    res andAlso andAlso
```

Comment:
one must sacrifice some poetry. Tomorrow, and tomorrow, and tomorrow, creeps in this petty pace from day to day...